packages/hardhat-core/test/internal/hardhat-network/console-log.ts

```diff
@@ -0,0 +1,77 @@
+import { assert } from "chai";
+import { useEnvironment } from "../../helpers/environment";
+import { useFixtureProject } from "../../helpers/project";
+import { TASK_COMPILE } from "../../../src/builtin-tasks/task-names";
+
+describe("Solidity console.log should print numbers without losing precision, occurrences of %d and %i should be correctly replaced with %s", function () {
+  const n1 = "11111111111111111111111111111111";
+  const n2 = "22222222222222222222222222222222";
+  const n3 = "33333333333333333333333333333333";
+
+  // Set up the test environment
+  useFixtureProject("console-log");
+  useEnvironment();
+
+  it("should print all the numbers without losing precision", async function () {
+    await this.env.run(TASK_COMPILE, { quiet: true });
+
+    // Retrieve the artifact of the solidity file compiled in the previous command
+    const artifact = this.env.artifacts.readArtifactSync("Test");
+
+    // Deploy contract and get receipt
+    const [deployer] = await this.env.network.provider.send("eth_accounts");
+    const tx = await this.env.network.provider.send("eth_sendTransaction", [
+      {
+        from: deployer,
+        data: artifact.bytecode,
+      },
+    ]);
+    const receipt = await this.env.network.provider.send(
+      "eth_getTransactionReceipt",
+      [tx]
+    );
+
+    // Modify console.log to store the messages that are gonna be printed when executing the smart contract function
+    const originalConsoleLog = console.log;
+    const capturedLogs: any[] = [];
+    console.log = (v: any) => {
+      capturedLogs.push(v);
+    };
+
+    // Call the contract function
+    await this.env.network.provider.send("eth_sendTransaction", [
+      {
+        from: deployer,
+        to: receipt.contractAddress,
+        data: "0xf8a8fd6d",
```

Comment:
I still think we should include this comment, or assign the string to a descriptive variable.

```suggestion
        // selector of 'test()'
        data: "0xf8a8fd6d",
```

Reply:
Done